### ISSUE TITLE
4.0b4 Keyboard Cursor Keys

### ISSUE THREAD

**Issue by bluecursor** [dirkwhoffmann/virtualc64#630]
I just tested on a real hardware with a joystick in port 2 and all the cursor keys work.

At boot and joystick in port 2 on, the arrow keys move the cursor except for arrow up. But SHiFT+ arrow UP was working, and now since enable/disabling "Disconnect emulation keys from keyboard" it doesn't work anymore.

There is also an issue with using the cursor keys with "Show Keyboard" Attached is a video od the issue

https://user-images.githubusercontent.com/31462938/105929509-bf993280-6015-11eb-8b38-31c80120e064.mp4

**Comment by dirkwhoffmann** [comment]
I've solved the bug described in the video (cursor moved more than one position). It was due to the way the virtual keyboard released keys (when a key was pressed, it scheduled a keyReleaseAll with an "async dispatch" on the Swift side). The new code is utilising the key event scheduler which is new in v4.

I don't understand the first part though. After

- plugging a joystick into port 1 and
- pulling up, left, right, down,

I see the following on the screen:

<img width="358" alt="Bildschirmfoto 2021-01-28 um 19 40 16" src="https://user-images.githubusercontent.com/12561945/106184417-c94fa100-61a1-11eb-8fa9-3fcb92ecee2a.png">

The same should happen on a real C64.
